azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -849,37 +824,27 @@ def _clear_reconnect_timer(self):
             self.reconnect_timer = None
 
     @pipeline_thread.runs_on_pipeline_thread
-    def on_connected(self):
-        """
-        Called by lower layers when the protocol client connects
-        """
-        logger.debug("{}: on_connected".format(self.name))
-
-        self._clear_reconnect_timer()
-
-        if self.previous:
-            self.previous.on_connected()
+    def _handle_pipeline_event(self, event):
+        if isinstance(event, pipeline_events_base.ConnectedEvent):
+            self._clear_reconnect_timer()
+            self.send_event_up(event)
 
-    @pipeline_thread.runs_on_pipeline_thread
-    def on_disconnected(self):
-        """
-        Called by lower layers when the protocol client disconnects
-        """
-        logger.debug("{}: on_disconnected".format(self.name))
-        if self.pipeline_root.connected:
-            if self.virtually_connected:
-                logger.info(
-                    "{}: disconnected but virtually connected.  Triggering reconnect timer.".format(
-                        self.name
+        elif isinstance(event, pipeline_events_base.DisconnectedEvent):
+            if self.pipeline_root.connected:
+                if self.virtually_connected:
+                    logger.info(
+                        "{}: disconnected but virtually connected.  Triggering reconnect timer.".format(
+                            self.name
+                        )
                     )
-                )
-                self._set_reconnect_timer()
-            else:
-                logger.info(
-                    "{}: disconnected, but not virtually connected.  Not triggering reconnect timer.".format(
-                        self.name
+                    self._set_reconnect_timer()
+                else:
+                    logger.info(
+                        "{}: disconnected, but not virtually connected.  Not triggering reconnect timer.".format(
+                            self.name
+                        )
                     )
-                )
+            self.send_event_up(event)
 
-        if self.previous:
-            self.previous.on_disconnected()
+        else:
+            super(ReconnectStage, self)._handle_pipeline_event(event)
```

Comment:
>super(ReconnectStage, self)._handle_pipeline_event(event) [](start = 12, length = 57)

To maintain similarity with the op_down for stages do we also need to do event_up for these ?